src/share.tsx

```diff
@@ -17,10 +19,65 @@ function Header(props: PropsWithChildren): JSX.Element {
   )
 }
 
-export function ShareSpace (): JSX.Element {
+function isDID(value: string): boolean {
+  try {
+    DID.parse(value.trim())
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+function isEmail(value: string): boolean {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+  return !isDID(value) && emailRegex.test(value)
+}
+
+export function ShareSpace({spaceDID}: {spaceDID: string}): JSX.Element {
   const [{ client }] = useW3()
   const [value, setValue] = useState('')
   const [downloadUrl, setDownloadUrl] = useState('')
+  const [sharedEmails, setSharedEmails] = useState<{ email: string, capabilities: string[] }[]>([])
+
+  useEffect(() => {
+    if (client && spaceDID) {
+      // Find all delegations where the spaceDID is present
+      const delegations = client.delegations()
+        .filter(d => d.capabilities.some(c => c.with === spaceDID))
+        .map(d => ({
+          email: DIDMailTo.toEmail(DIDMailTo.fromString(d.audience.did())),
+          capabilities: d.capabilities.map(c => c.can)
+        }))
+      setSharedEmails(delegations)
+    }
+  }, [client, spaceDID])
+
+  async function shareViaEmail(email: string): Promise<void> {
+    try {
+      if (!client) {
+        throw new Error(`Client not found`)
+      }
+
+      const space = client.spaces().find(s => s.did() === spaceDID)
+      if (!space) {
+        throw new Error(`Could not find space to share`)
+      }
+
+      const delegatedEmail = DIDMailTo.email(email)
+      const delegation = await client.shareSpace(delegatedEmail, space.did())
+
+      const next = { email: delegatedEmail, capabilities: delegation.capabilities.map(c => c.can) }
+      setSharedEmails(prev => {
+        if (prev.some(item => item.email === next.email)) {
+          return prev
+        }
+        return [...prev, next]
```

Comment:
Instead of appending to the existing list, could you just make this a function that you call from the effect above as well as here?